deepchem/models/torch_models/mpnn.py

```diff
@@ -0,0 +1,323 @@
+
+    Returns
+    -------
+    torch.Tensor
+      The model output.
+
+      * When self.mode = 'regression',
+        its shape will be ``(dgl_graph.batch_size, self.n_tasks)``.
+      * When self.mode = 'classification', the output consists of probabilities
+        for classes. Its shape will be
+        ``(dgl_graph.batch_size, self.n_tasks, self.n_classes)`` if self.n_tasks > 1;
+        its shape will be ``(dgl_graph.batch_size, self.n_classes)`` if self.n_tasks is 1.
+    torch.Tensor, optional
+      This is only returned when self.mode = 'classification', the output consists of the
+      logits for classes before softmax.
+    """
+    node_feats = g.ndata[self.nfeat_name]
+    edge_feats = g.edata[self.efeat_name]
+    out = self.model(g, node_feats, edge_feats)
+
+    if self.mode == 'classification':
+      if self.n_tasks == 1:
+        logits = out.view(-1, self.n_classes)
+        softmax_dim = 1
+      else:
+        logits = out.view(-1, self.n_tasks, self.n_classes)
+        softmax_dim = 2
+      proba = F.softmax(logits, dim=softmax_dim)
+      return proba, logits
+    else:
+      return out
+
+
+class MPNNModel(TorchModel):
+  """Model for graph property prediction
+
+  This model proceeds as follows:
+
+  * Combine latest node representations and edge features in updating node representations,
+    which involves multiple rounds of message passing
+  * For each graph, compute its representation by combining the representations
+    of all nodes in it, which involves a Set2Set layer.
+  * Perform the final prediction using an MLP
+
+  Examples
+  --------
+
+  >>>
+  >> import deepchem as dc
+  >> from deepchem.models.torch_models import MPNNModel
+  >> featurizer = dc.feat.MolGraphConvFeaturizer(use_edges=True)
+  >> tasks, datasets, transformers = dc.molnet.load_tox21(
+  ..     reload=False, featurizer=featurizer, transformers=[])
+  >> train, valid, test = datasets
+  >> model = MPNNModel(mode='classification', n_tasks=len(tasks),
+  ..                   batch_size=32, learning_rate=0.001)
+  >> model.fit(train, nb_epoch=50)
+
+  References
+  ----------
+  .. [1] Justin Gilmer, Samuel S. Schoenholz, Patrick F. Riley, Oriol Vinyals, George E. Dahl.
+         "Neural Message Passing for Quantum Chemistry." ICML 2017.
+
+  Notes
+  -----
+  This class requires DGL (https://github.com/dmlc/dgl) and DGL-LifeSci
+  (https://github.com/awslabs/dgl-lifesci) to be installed.
+  """
+
+  def __init__(self,
+               n_tasks: int,
+               node_out_feats: int = 64,
+               edge_hidden_feats: int = 128,
+               num_step_message_passing: int = 3,
+               num_step_set2set: int = 6,
+               num_layer_set2set: int = 3,
+               mode: str = 'regression',
+               number_atom_features: int = 30,
+               number_bond_features: int = 11,
+               n_classes: int = 2,
+               nfeat_name: str = 'x',
+               efeat_name: str = 'edge_attr',
```

Comment:
Do these arguments serve any purpose?  `_prepare_batch()` expects the features to be GraphData objects, which it calls `to_dgl_graph()` on.  That method hardcodes the names "x" and "edge_attr".  So I don't see any benefit in making them configurable here.

Reply:
You are right. I've fixed it.